File: new-algorithm/src/javaalgo/programmers/level1_콜라츠_추측.java
package javaalgo.programmers;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class level1_콜라츠_추측 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        System.out.println(solution(n));
    }

    static int solution(long num) {
        int cnt = 0;

        while (num != 1) {

            if (num % 2 == 0) {
                num /= 2;
            } else {
                num = num * 3 + 1;
            }
            cnt++;

            if (cnt == 500) {
                cnt = -1;
                break;
            }
        }

        return cnt;
    }
}

File: new-algorithm/src/kotlinalgo/programmers/level1_콜라츠_추측.kt
package kotlinalgo.programmers

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val n = Integer.parseInt(readLine())
    println(solution_콜라츠(n))
}

fun solution_콜라츠(num: Int): Int {
    var tmp = num.toLong()
    var cnt = 0;

    while (cnt < 500 && tmp > 1) {
        cnt++
        tmp = if (tmp % 2 == 0L) {
            tmp / 2
        } else {
            tmp * 3 + 1
        }
    }

    return if (cnt == 500)  -1 else cnt
}
